### PR TITLE
CWU Vendor Proposal Cypress Test

### PR DESCRIPTION
**Pull request by BCerki**
This PR addresses issue: #134 

Includes tests? y
Rebased on main address to conflicts? y
Updated docs? n/a

Proposed changes:
- update test authentication to allow for vendor roles
- write a test for a vendor creating and reading a CWU proposal
- write a test for a vendor updating and deleting a CWU proposal
- add fixtures for ^ tests
- update existing tests to use better selectors I discovered while writing the vendor CWU tests

### CODE REVIEW

File: src/back-end/lib/routers/auth.ts [hunk 1]
```diff
@@ -136,11 +137,11 @@ async function makeRouter(connection: Connection): Promise<Router<any, any, any,
         }
       })
     },
-    // /auth/createsession is for testing only, so only exists outside of production
+    // /auth/createsessiongov is for testing only, so only exists outside of production
     // @ts-ignore
     ...(process.env.NODE_ENV === 'development' ? [{
       method: ServerHttpMethod.Get,
-      path: '/auth/createsession',
+      path: '/auth/createsessiongov',
```
Comment (BCerki) [inline src/back-end/lib/routers/auth.ts:144]: I couldn't think of a way to use the same endpoint to authenticate different users. I thought of env variables, but [Cypress doesn't seem to be able to change OS-level variables ](https://docs.cypress.io/guides/guides/environment-variables#:~:text=Difference%20between%20OS%2Dlevel%20and,from%20OS%2Dlevel%20environment%20variables.)

File: cypress/integration/vendor-cwu-CR.spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        cy.login('vendor')
+        cy.sqlFixture('cwuOpportunity.sql')
+    })
+
+    it('create and read CWU proposal', function() {
+        cy.get('#user-sign-up-step-two-terms').click()
+        cy.get('a').contains('Complete Profile').click()
+        cy.get('a[href="/opportunities"]').contains('View All Opportunities').click()
+        cy.contains('Fixture CWU Opportunity Title').should('be.visible')
+        cy.contains('Fixture CWU Opportunity Title').click()
```
Comment (BCerki) [inline cypress/integration/vendor-cwu-CR.spec.js:16]: Putting an assertion before an attempt to click generally solves the local vs. CI timing issues. I only put it in the places where I had problems, not everywhere in the test.

File: cypress/integration/vendor-cwu-UD.spec.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        cy.sqlFixture('dbReset.sql')
+        cy.login('vendor')
+        cy.sqlFixture('cwuOpportunity.sql')
+        cy.sqlFixture('cwuProposal.sql')
+    })
+
+    it('edit CWU proposal', function() {
+        cy.get('#user-sign-up-step-two-terms').click()
+        cy.get('a').contains('Complete Profile').click()
+        cy.visit('/opportunities')
```
Comment (BCerki) [inline cypress/integration/vendor-cwu-UD.spec.js:14]: There's a button you can click, but I was having a lot of timeout issues around it, so I did this instead and it seemed (?) to help

File: cypress/integration/vendor-cwu-CR.spec.js (hunk 2 above)
Comment (mikevespi) [reply cypress/integration/vendor-cwu-CR.spec.js:16]: great!

File: src/back-end/lib/routers/auth.ts (hunk 1 above)
Comment (mikevespi) [reply src/back-end/lib/routers/auth.ts:144]: That's fine, your solution works

**Review by mikevespi — APPROVED** [review]
Great stuff 🚢

**Pull request merged**